code/Control/UserDefinedFormController.php

```diff
@@ -315,8 +315,8 @@ public function process($data, $form)
                             $field->createProtectedFolder();
                         }
 
-                        $file = Versioned::withVersionedMode(function () use ($field, $form) {
-                            $stage = Injector::inst()->get(self::class)->config()->get('file_upload_stage');
+                        $stage = $this->config()->get('file_upload_stage');
+                        $file = Versioned::withVersionedMode(function () use ($field, $form, $stage) {                    
```

Comment:
```suggestion
                        $file = Versioned::withVersionedMode(function () use ($field, $form, $stage) {
```
This is the change the phplinter wants you to make